alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteWorkflowAction.kt

```diff
@@ -0,0 +1,255 @@
+import org.opensearch.common.xcontent.XContentHelper
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentType
+import org.opensearch.commons.alerting.AlertingPluginInterface
+import org.opensearch.commons.alerting.action.AlertingActions
+import org.opensearch.commons.alerting.action.DeleteMonitorRequest
+import org.opensearch.commons.alerting.action.DeleteMonitorResponse
+import org.opensearch.commons.alerting.action.DeleteWorkflowRequest
+import org.opensearch.commons.alerting.action.DeleteWorkflowResponse
+import org.opensearch.commons.alerting.model.CompositeInput
+import org.opensearch.commons.alerting.model.ScheduledJob
+import org.opensearch.commons.alerting.model.Workflow
+import org.opensearch.commons.authuser.User
+import org.opensearch.commons.utils.recreateObject
+import org.opensearch.index.IndexNotFoundException
+import org.opensearch.index.query.QueryBuilders
+import org.opensearch.rest.RestStatus
+import org.opensearch.search.builder.SearchSourceBuilder
+import org.opensearch.tasks.Task
+import org.opensearch.transport.TransportService
+
+/**
+ * Transport class that deletes the workflow.
+ * If the deleteDelegateMonitor flag is set to true, deletes the workflow delegates that are not part of another workflow
+ */
+class TransportDeleteWorkflowAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters,
+    val clusterService: ClusterService,
+    settings: Settings,
+    val xContentRegistry: NamedXContentRegistry
+) : HandledTransportAction<ActionRequest, DeleteWorkflowResponse>(
+    AlertingActions.DELETE_WORKFLOW_ACTION_NAME, transportService, actionFilters, ::DeleteWorkflowRequest
+),
+    SecureTransportAction {
+
+    private val log = LogManager.getLogger(javaClass)
+
+    @Volatile override var filterByEnabled = AlertingSettings.FILTER_BY_BACKEND_ROLES.get(settings)
+
+    init {
+        listenFilterBySettingChange(clusterService)
+    }
+
+    override fun doExecute(task: Task, request: ActionRequest, actionListener: ActionListener<DeleteWorkflowResponse>) {
+        val transformedRequest = request as? DeleteWorkflowRequest
+            ?: recreateObject(request) { DeleteWorkflowRequest(it) }
+
+        val user = readUserFromThreadContext(client)
+        val deleteRequest = DeleteRequest(ScheduledJob.SCHEDULED_JOBS_INDEX, transformedRequest.workflowId)
+            .setRefreshPolicy(RefreshPolicy.IMMEDIATE)
+
+        if (!validateUserBackendRoles(user, actionListener)) {
+            return
+        }
+
+        GlobalScope.launch(Dispatchers.IO + CoroutineName("DeleteWorkflowAction")) {
+            DeleteWorkflowHandler(
+                client,
+                actionListener,
+                deleteRequest,
+                transformedRequest.deleteDelegateMonitors,
+                user,
+                transformedRequest.workflowId
+            ).resolveUserAndStart()
```

Comment:
do we need make `resolveUserAndStart` a `suspend` function? is there any advantage? i see `TransportIndexMonitorAction` not doing it. https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt#L247

Reply:
deleteMonitors and deleteWorkflow are suspend functions - both are called by resolveUserAndStart